content/sensu-go/6.8/catalog/catalog-reference.md

```diff
@@ -0,0 +1,1235 @@
+This makes it possible to give users access to earlier versions of integrations to hedge against regressions in individual integrations.
+
+For example, in the [official Sensu Catalog repository][30], two `ansible-tower-remediation` version tags are defined:
+
+{{< code text >}}
+git tag --list |grep ansible-tower-remediation
+ansible/ansible-tower-remediation/20220223.0.0
+ansible/ansible-tower-remediation/20220421.0.0
+{{< /code >}}
+
+Using these tags, the catalog-api tool will generate the following integration structure:
+
+{{< code text >}}
+tree /tmp/generated-api/ -L 7
+/tmp/generated-api/
+├── release
+│   ├── 5029648381dff2426ea247147456b4f1227fd6d9050fa42f0660e67a218f8c87
+│   │   └── v1
+│   │ ├── ansible
+│   │ │   ├── ansible-tower-remediation
+│   │ │   │   ├── 20220223.0.0
+│   │ │   │   │   ├── CHANGELOG.md
+│   │ │   │   │   ├── img
+│   │ │   │   │   ├── logo.png
+│   │ │   │   │   ├── README.md
+│   │ │   │   │   └── sensu-resources.json
+│   │ │   │   ├── 20220223.0.0.json
+│   │ │   │   ├── 20220421.0.0
+│   │ │   │   │   ├── CHANGELOG.md
+│   │ │   │   │   ├── img
+│   │ │   │   │   ├── logo.png
+│   │ │   │   │   ├── README.md
+│   │ │   │   │   └── sensu-resources.json
+│   │ │   │   ├── 20220421.0.0.json
+│   │ │   │   └── versions.json
+│   │ │   └── ansible-tower-remediation.json
+{{< /code >}}
+
+## Use the Sensu Catalog API server during integration development
+
+When you're developing integrations, it can be helpful to run the Sensu Catalog API server from your local environment so that you can preview integrations as you work.
+To do this, use the `server` subcommand in the catalog-api command line tool.
+
+{{% notice note %}}
+**NOTE**: Make sure you have a local Sensu instance running with access to the Sensu web UI.
+{{% /notice %}}
+
+1. Clone the Sensu Catalog API repository:
+{{< code shell >}}
+git clone https://github.com/sensu/catalog-api
+{{< /code >}}
+
+2. Navigate to the local catalog-api repository:
+{{< code shell >}}
+cd catalog-api
+{{< /code >}}
+
+3. Build the `catalog-api` tool:
+{{< code shell >}}
+go build
+{{< /code >}}
+
+4. Clone the repository that stores your Sensu integrations.
+This example uses Sensu's public integration repository:
+{{< code shell >}}
+git clone https://github.com/sensu/catalog
+{{< /code >}}
+
+5. Navigate to your local copy of the repository that stores the Sensu integrations.
+This example uses https://github.com/sensu/catalog, so the repository is `catalog`: 
+{{< code shell >}}
+cd ../catalog
+{{< /code >}}
+
+6. Run the catalog-api `server` subcommand:
+{{< code shell >}}
+../catalog-api/catalog-api catalog server --repo-dir . -watch
+{{< /code >}}
+
+   The `.` in the command tells Sensu to read the catalog contents from your local environment.
+   Use the `-watch` flag to reload the API as you save updates in integration files so that you can see them live in the Sensu web UI.
```

Comment:
I'm not sure if this command is correct

Reply:
There seems to be another step here about adding two headers: one to override CORS and the other to prevent caching the browser content. Jef does this with Python in the customer demo, but I'm not sure how users should do this.

Reply:
The server in the catalog-api tool deals with CORS & caching for us. Users will only need to set them if they're serving the catalog api using a different webserver.

Reply:
Can you expand on this a bit? I'm not sure what you mean by a different webserver. Jef showed it in a customer demo, so I'm thinking we have at least one customer who wants or needs to do this.

Reply:
The catalog-api tool can perform two main tasks:
1. It can look at a Catalog Git repository & generate a Catalog API. A Catalog API is a hierarchical structure of directories, JSON files, YML files, and images. These directories & files are useless by themselves; they need a web server to give the world (or other users) access to them through HTTP/HTTPS.
2. A web server that serves the generated Catalog API.

The web server built into the catalog-api tool should only be used for local development. Users should use a solution such as Nginx, Apache, Amazon S3/Cloudfront, etc. The instructions on configuring CORS and caching is dependent on which software the user is using to serve the Catalog API.

Reply:
Ah ok, that makes sense. Thank you.